apps/ledger-live-mobile/src/screens/FirmwareUpdate/index.tsx

```diff
@@ -165,6 +173,19 @@ export const FirmwareUpdate = ({
   const productName = getDeviceModel(device.modelId).productName;
 
   const [fullUpdateComplete, setFullUpdateComplete] = useState(false);
+  const [showBatteryWarningDrawer, setShowBatteryWarningDrawer] = useState<boolean>(false);
+  const [showReleaseNotes, setShowReleaseNotes] = useState<boolean>(true);
+
+  const {
+    requestCompleted: batteryRequestCompleted,
+    batteryStatusesState,
+    triggerRequest: triggerBatteryCheck,
+    cancelRequest: cancelBatteryCheck,
+    isBatteryLow,
+  } = useBatteryStatuses({
```

Comment:
We'll need to see how we can handle this for the nano X when we enable the fw update via BLE

BUT actually: this screen/flow is used when updating a nano device via USB on Android ! Not sure it works with nano SP and nano X
(i'm almost sure the APDU does not exist on them, or at least on older fw versions)

You should double check, and if it does not work, either ignore the error or disable the hook for them

Reply:
And maybe it's better to handle this business logic: "is the battery check available on this current device" be handled on live-common in the action/hook (and not here in this component). It could return a state saying that the battery check is not available

Reply:
The new nano X respond correctly to the APDU with a firmware of `v2.2.2`. The hook handle the battery correctly and notify that the device is plugged in. In fact, there is no battery warning.

Therefore, on older firmware, such as `v2.0.0` this APDU does not exist. So, I need to ensure that the drawer doesn't block the flow.

Reply:
With new modifications, older firmware are now responding correctly